### PR TITLE
#45 Use underline text to indicate clickable words

### PR DESCRIPTION
**Pull request by gscdev**

### CODE REVIEW

File: app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java [hunk 1]
```diff
@@ -194,6 +196,12 @@ public void onClick(@NonNull View widget) {
                                     // Report learning event to the Analytics application (https://github.com/elimu-ai/analytics)
                                     LearningEventUtil.reportWordLearningEvent(word, LearningEventType.WORD_PRESSED, getContext(), BuildConfig.ANALYTICS_APPLICATION_ID);
                                 }
+
+                                @Override
+                                public void updateDrawState(@NonNull TextPaint ds) {
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java:201]: Rename `ds` to something more descriptive? I wasn't able to understand this abbreviation.

File: app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java [hunk 2]
```diff
@@ -194,6 +196,12 @@ public void onClick(@NonNull View widget) {
...
+                                    ds.setColor(Color.BLACK);
+                                    ds.setUnderlineText(true);
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java:203]: Is it possible to set the color of the underline itself? If so, perhaps we could use something like `colorPrimary` or `colorAccent` in `colors.xml` as the color of the line (to make it a little less visible).

device-2021-01-09-021915

**Review by nya-elimuai — CHANGES_REQUESTED** [review]

File: app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java (hunk 2 above)
Comment (gscdev) [reply app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java]: @nya-elimuai I've changed the underline color

device-2021-01-15-003042

File: app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java [hunk 3]
```diff
@@ -312,4 +308,4 @@ public void onError(String utteranceId) {
     @Override
     public void onAudioDone() {
     }
-}
+}
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java:311]: Add newline at the end of the file

File: app/src/main/java/ai/elimu/vitabu/util/ColoredUnderlineSpan.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+        canvas.drawText(text, start, end, x, y, paint);
+        canvas.drawRect(x, lineBottom - thickness, (x + paint.measureText(text, start, end)), lineBottom, linePaint);
+    }
+
+    @Override
+    public int getSize(@NonNull Paint paint, CharSequence text, int start, int end, @Nullable Paint.FontMetricsInt fm) {
+        return (int) paint.measureText(text, start, end);
+    }
+}
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/vitabu/util/ColoredUnderlineSpan.java:37]: Add newline at the end of the file

**Review by nya-elimuai — DISMISSED** [review]
Looks great! 😄

device-2021-01-15-213840

device-2021-01-15-213921

**Review by jo-elimuai — DISMISSED** [review]

File: app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java [hunk 5]
```diff
@@ -304,4 +308,4 @@ public void onError(String utteranceId) {
...
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.java:311]:
```suggestion
}

```

**Review by jo-elimuai — APPROVED** [review]

**Review by nya-elimuai — APPROVED** [review]

**Pull request merged**